src/ModelContextProtocol/Protocol/Transport/StdioServerTransport.cs

```diff
@@ -83,16 +84,57 @@ public StdioServerTransport(string serverName, ILoggerFactory? loggerFactory = n
         
         _serverName = serverName;
         _logger = (ILogger?)loggerFactory?.CreateLogger<StdioClientTransport>() ?? NullLogger.Instance;
+        
+        // Create console streams with explicit UTF-8 encoding to ensure proper Unicode character handling
+        // This is especially important for non-ASCII characters like Chinese text and emoji
+        var utf8Encoding = new UTF8Encoding(false); // No BOM
+        
+        // Get raw console streams and wrap them with UTF-8 encoding
+        Stream inputStream = Console.OpenStandardInput();
+        Stream outputStream = Console.OpenStandardOutput();
+        
+        _stdInReader = new StreamReader(inputStream, utf8Encoding);
+        _stdOutWriter = new StreamWriter(outputStream, utf8Encoding) { AutoFlush = true };
```

Comment:
If we're going to use Console.OpenStandardOutput, we don't actually need a writer at all. At that point we can just write directly to the stream, which also means the JsonSerializer calls can serialize directly to the stream, rather than going through an intermediate string.